### ISSUE TITLE
Can't upload same version number for multiple architectures

### ISSUE THREAD

**Issue by Arc676** [UbuntuOpenStore/openstore-meta#215]
I wrote an app for Ubuntu Touch using `clickable` and QML, but I have click packages both for desktop (amd64) and mobile (armhf). However, OpenStore won't allow me to upload both under the same version number.

Since the primary target was mobile and it's easier to compile on desktop, my current workaround is to up the version to 0.1.1 to upload the armhf build and just provide a link to the repository for users to download the desktop version.

However, I can't find a link to older versions, suggesting users can only download the latest version of any application. This means that the amd64 version of the app is now unavailable to users who might not want to compile it themselves.

The alternative solution is to build the same app twice but under different identifiers and have two OpenStore pages.

**Comment by bhdouglass** [comment]
Generally apps on the openstore only target armhf for mobile, but if you want to publish both you can combine them into one click package that contains the binaries for both armhf and amd64.

**Comment by Arc676** [comment]
Thanks for the reply.

Is this possible using `clickable`? I didn't see a mention of this in the `man` page.

Wouldn't combining the packages mean double the download size though? I realize that this is negligible for small apps, but fundamentally the user would be downloading around twice the needed size to only use half of it. Or does the clickable system have something in place for this?

**Comment by bhdouglass** [comment]
No, clickable doesn't support this right now since the only ubuntu touch devices are armhf.

Are you able to install your amd64 click on a desktop?

**Comment by Arc676** [comment]
I only ever tested on desktop using `clickable -e`.

I don't actually remember what tool to use to install click packages.

**Comment by bhdouglass** [comment]
Last I knew clicks don't really have good support on the desktop. You might want to look into creating a snap for the desktop. I think there is some sort of framework snap for the Ubuntu sdk that would be useful.
